src/librustdoc/doctest.rs

```diff
@@ -501,11 +548,12 @@ fn run_test(
     rustdoc_options: &RustdocOptions,
     supports_color: bool,
     report_unused_externs: impl Fn(UnusedExterns),
-) -> Result<(), TestFailure> {
+) -> (Duration, Result<(), TestFailure>) {
```

Comment:
There should be a comment explaining that this is the time taken for *compilation*, not the total time overall.